cli/deployment/config.go

```diff
@@ -134,24 +135,52 @@ func (c Config) YAML(outputs map[string]map[string]interface{}) ([]byte, error)
 
 // Function findAllDependencies finds all dependencies base on cross-deployment references found in config YAML,
 // if no dependencies found, returns empty slice.
-func (c Config) findAllDependencies(configs map[string]Config) []Config {
+func (c Config) findAllDependencies(configs map[string]Config) ([]Config, error) {
 	var result []Config
 	refs := c.findAllOutRefs()
 	dependencies := map[string]Config{}
 	for _, ref := range refs {
 		fullName, _, _ := c.parseOutRef(ref)
 		dependency, found := configs[fullName]
+		exists := false
 		if !found {
-			log.Fatalf("Could not find config for deployment = %s", fullName)
+			projectAndName := strings.Split(fullName, ".")
+			deployment := &Deployment{
+				config: Config{
+					Project: projectAndName[0],
+					Name:    projectAndName[1],
+				},
+			}
+			status, err := GetStatus(deployment)
+
+			if err != nil {
+				log.Printf("GetStatus for deployment = %s", fullName)
+				return nil, err
+			}
+			switch status {
+			case Done:
+				exists = true
+			case NotFound:
+				message := fmt.Sprintf("Could not find config or existing deployment = %s", fullName)
+				log.Print(message)
+				return nil, errors.New(message)
+			case Pending, Error, Running:
+				message := fmt.Sprintf("Dependency deployment = %s is in PENDING, RUNNING or ERROR state", fullName)
```

Comment:
we should say what state is actually is in.

Reply:
will do

Reply:
@morgante this is good example of case why I put huge commit in past, the code you commented related to another PR https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/350, but I need this commit as well in this PR, so we have bit mess in result, I push fix for your comment in #350 PR

Reply:
If this PR depends on #350, you should just mark it as such and based on 350.

Reply:
If this PR depends on #350, you should just mark it as such and based on 350.